lib/ioutil/log_reader.go

```diff
@@ -24,12 +24,9 @@ import (
 // LogReader unwraps entires into LogEntry structs.
 type LogReader struct {
 	io.ReadCloser
-	s *bufio.Scanner
 
-	// if the buffer we want to read into is too short for our log entry,
-	// we will need to return io.ErrShortBuffer and try again on the next call,
-	// necessitating this pointer to the previous LogEntry we attempted to use.
-	prev *LogEntry
+	s    *bufio.Scanner
+	prev *string
```

Comment:
I am not sure you will get empty strings here. Since any string you get will at least have "\n" at the end which is already means it is not empty.